sql/src/test/java/io/crate/execution/engine/window/WindowFunctionTest.java

```diff
@@ -0,0 +1,170 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+
+class WindowFunctionTest {
+
+    private SqlExpressions sqlExpressions;
+    private Functions functions;
+    private Map<QualifiedName, AnalyzedRelation> tableSources;
+    private TransactionContext txnCtx = CoordinatorTxnCtx.systemTransactionContext();
+    private InputFactory inputFactory;
+
+    private final RamAccountingContext RAM_ACCOUNTING_CONTEXT =
+        new RamAccountingContext("dummy", new NoopCircuitBreaker("dummy"));
+
+    @Before
+    public void prepareFunctions() throws Exception {
+        DocTableInfo tableInfo = TestingTableInfo.builder(new RelationName(DocSchemaInfo.NAME, "users"), null)
+            .add("id", DataTypes.INTEGER)
+            .add("name", DataTypes.STRING)
+            .add("tags", new ArrayType(DataTypes.STRING))
+            .add("age", DataTypes.INTEGER)
+            .add("a", DataTypes.INTEGER)
+            .add("x", DataTypes.LONG)
+            .add("shape", DataTypes.GEO_SHAPE)
+            .add("timestamp", DataTypes.TIMESTAMP)
+            .add("timezone", DataTypes.STRING)
+            .add("interval", DataTypes.STRING)
+            .add("time_format", DataTypes.STRING)
+            .add("long_array", new ArrayType(DataTypes.LONG))
+            .add("int_array", new ArrayType(DataTypes.INTEGER))
+            .add("array_string_array", new ArrayType(new ArrayType(DataTypes.STRING)))
+            .add("long_set", new SetType(DataTypes.LONG))
+            .add("regex_pattern", DataTypes.STRING)
+            .add("geoshape", DataTypes.GEO_SHAPE)
+            .add("geopoint", DataTypes.GEO_POINT)
+            .add("geostring", DataTypes.STRING)
+            .add("is_awesome", DataTypes.BOOLEAN)
+            .add("double_val", DataTypes.DOUBLE)
+            .add("float_val", DataTypes.DOUBLE)
+            .add("short_val", DataTypes.SHORT)
+            .add("obj", DataTypes.OBJECT, ImmutableList.of())
+            .build();
+        DocTableRelation tableRelation = new DocTableRelation(tableInfo);
+        tableSources = ImmutableMap.of(new QualifiedName("users"), tableRelation);
+        sqlExpressions = new SqlExpressions(tableSources);
+        functions = sqlExpressions.functions();
+        inputFactory = new InputFactory(functions);
+    }
+
+
+    private void assertInputAndExpectedSize(Map<String, List<Literal>> inputValueMap, Object[] expectedValues) {
+        assertThat("No Input Provided", inputValueMap.isEmpty(), is(false));
+
+        List<Integer> inputSizes = inputValueMap.values()
+            .stream()
+            .map(l -> l.size())
+            .distinct()
+            .collect(Collectors.toList()
+            );
+        assertThat("Input Lists need to be of equal size",
```

Comment:
Not sure if I'd use `assertThat` for pre-condition checks of the testing API. I think we should distinguish between "test failures" (function in question not returning the correct result) and "incorrect test framework API usage"). So I'd throw regular IllegalArgumentExceptions here.